### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by ismailakinkunmi**
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
      If response status was correct.
      If a correct template was rendered.
      If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    it 'render show.html.erb view' do
+      get '/users/1'
+      expect(response).to render_template('show')
+    end
+
+    it 'should render a user by id' do
+      get '/users/1'
+      expect(response.body).to include('Get user by id')
+    end
+  end
```
Comment (TedLivist) [inline spec/requests/users_spec.rb:27]: - [ ] Your user show tests are failing. Kindly fi them.

  show-user

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    it "should render 'show.html.erb' view" do
+      get '/users/1/posts/1'
+      expect(response.body).to include('Get post by id')
+    end
+
+    it "should render 'show.html.erb' view" do
+      get '/users/1/posts/1'
+      expect(response.body).to render_template('show')
+    end
+  end
```
Comment (TedLivist) [inline spec/requests/posts_spec.rb:27]: - [ ] Your post show tests are failing. Kindly fix them.

  post-show

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ismailakinkunmi,

Good job so far!
✔️ Good documentation
✔️ Index tests are passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (ismailakinkunmi) [reply spec/requests/posts_spec.rb:27]: @TedLivist thanks for the review and i tried to run the test again all are working 
Note: incase of next reviewer if there is a fail case kindly run by line I.e  rspec ./spec/requests/users_spec.rb: 7 to help me locate the fail test thanks all.


Screenshot (191)

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @ismailakinkunmi**, 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Has `users_spec.rb` and `posts_spec.rb`, keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] It would be nice to add explicit steps on how to run tests inside your README file.

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ismailakinkunmi** [comment]
@BenMukebo noted and thank you.

**Pull request merged**